pandas/io/stata.py

```diff
@@ -1646,8 +1647,14 @@ def _do_convert_missing(self, data, convert_missing):
                     dtype = np.float64
                 replacement = Series(series, dtype=dtype)
                 replacement[missing] = np.nan
-
-            data[colname] = replacement
+            replacements[colname] = replacement
+        if replacements:
+            columns = data.columns
+            replacements = DataFrame(replacements)
+            data.drop(replacements.columns, 1, inplace=True)
```

Comment:
don't use inplace

Reply:
I used `inplace` intentionally to try and minimize memory consumption, which can be an issue when loading some Stata files. Is it going away?

Reply:
its not idiomatic, nor does it actually do anything here (it rarely does)